### DISCUSSION TITLE
Feature Request: Rest API support for all Enterprise Settings

### DISCUSSION THREAD

**Discussion by HariSekhon** [community/community#19570] [Enterprise]
Feature Request for the Rest API to support all enterprise settings available in the UI, such as:

- disabling repository creation by members
- repository forking
- outside collaborators
- GitHub Actions:
  - 3rd party actions and workflows allowed (can currently only be done at organization level)
  - read-only workflow permissions by default
- repository admin permissions for:
  - repo visibility change
  - deletion and transfer
  - issue deletion
- Biliing spending limits

Use cases include scripting, automation and Terraform management.

Related ticket for Terraform GitHub provider which is dependent on this:

https://github.com/integrations/terraform-provider-github/issues/1199

**Comment by ausfestivus** [comment]
Thanks for raising this @HariSekhon. As a soon to be GitHub customer who uses Terraform extensively, the lack of an API for the Enterprise configuration items is disappointing.

We are a small team of nine. We operate hundreds of cloud services across the three major platforms under a tightly regulated environment. The services we provide to the business are consumed by thousands of staff. Some of those staff produce the most amazing public-facing cloud-based capabilities that are used around the world.

Our staff are crying out for us to take on GitHub and make it available for their use. We're keen to do it. The lack of this API (and hence no Terraform Provider support) will make our lives that much harder. Not just technically but also from a Governance and Assurance perspective.

Very surprising to learn that GitHub released a product/feature without a corresponding API for its use.
